### PR TITLE
Add Summit-XL Steel robot

### PR DESCRIPTION
**Pull request by mael25**
Adds the Robotnik Summit-XL Steel robot, alongside the Sick S300 lidar that it uses.

**Tasks**
  - [x] Create Icons
  - [x] Documentation Summit-XL Steel
  - [x] Better appearance & documentation SickS300 (lidar-sensors.md)
  - [x] Update robots Makefile
  - [x] Update Changelog
  - [x] Improve appearance base & Redo pictures/icons
  - [x] Add web component
 
summit_xls

### CODE REVIEW

File: docs/guide/summit-xl-steel.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+More information on the Summit-XL Steel robot can be found on the [constructor website](https://robotnik.eu/products/mobile-robots/summit-xl-steel-en/).
+
+### SummitXlSteel PROTO
+
+Derived from [Robot](../reference/robot.md).
+
+```
+SummitXlSteel {
+  SFVec3f     translation     0 0.118 0
+  SFRotation  rotation        1 0 0 -1.5708
```
Comment (omichel) [inline docs/guide/summit-xl-steel.md:16]: The default rotation should be along the Z axis, not X. Also the angle value should be 0.

File: docs/guide/summit-xl-steel.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+The model is a modular mobile platform used for logistics and indoor transport. It is well-suited for industries, as its design enables it to carry a payload of up to 250 kg.
+More information on the Summit-XL Steel robot can be found on the [constructor website](https://robotnik.eu/products/mobile-robots/summit-xl-steel-en/).
+
+### SummitXlSteel PROTO
+
+Derived from [Robot](../reference/robot.md).
+
+```
+SummitXlSteel {
+  SFVec3f     translation     0 0.118 0
```
Comment (omichel) [inline docs/guide/summit-xl-steel.md:15]: The orientation of the robot should be FLU, hence its default translation should be 0 0 0.118.

File: projects/devices/sick/protos/SickS300.proto [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+#  - range: up to 30 [m] (distance measuring range)
+#  - number of layers: 3
+#  - angular resolution: 0.5°
+#  - resolution: 540
+#  - dimension: 0.102 x 0.152 x 0.106 [m]
+#  - weight: 1.2 [kg]
+
+PROTO SickS300 [
+field SFVec3f    translation    0 0 0
+field SFRotation rotation       0 1 0 0
```
Comment (omichel) [inline projects/devices/sick/protos/SickS300.proto:17]:
```suggestion
field SFRotation rotation       0 0 1 0
```

**Review by omichel — COMMENTED** [review]
From the picture of the sick sensor, it looks like the mesh has no crease angle as we can see the faces (normal per face) on the cylinder-like part.

**Comment by mael25** [comment]
> From the picture of the sick sensor, it looks like the mesh has no crease angle as we can see the faces (normal per face) on the cylinder-like part.

Yes, I checked this and it was due to the structure of the mesh: there were thin faces at each "corner" in addition to the bigger faces. I have corrected this and it works well now.
sicks300

**Comment by mael25** [comment]
I have added the web component, the PR should be ready now. I have kept the `rotation 0 1 0 0` field for the SickS300 as the Lidars are not yet in FLU.

**Comment by mael25** [comment]
I just thought it might be better to put the keyboard controller of the ConveyorPlatform directly in its controllers folder rather than in the one of the Summit-XL Steel

**Review by BenjaminDeleze — CHANGES_REQUESTED** [review]
Is it possible to move the viewpoint back a bit in the .x3d file. Because if I open it in the documentation viewer I can only see the wheels and I have to zoom out.

**Comment by mael25** [comment]
This PR has been split in two other that target now the `develop` branch: #3121 for the `Summit-XL Steel` robot and #3122 for the `Sick S300` Lidar

**Pull request closed without merging**